### PR TITLE
Use SSIM image comparison

### PR DESCRIPTION
**Pull request by mwestphal**
- Use vtkImageSSIM when available in the provided VTK version
 - Rework image::compare API by using a better error metrics and drop the diff image
 - Update all baselines that were incorrect and not detected as such because of the previous algorithm
 - Add legacy support for previous VTK version
 - Exclude a few more tests for macOS and sanitizer
 - Update usages in F3D and libf3d
 - Update testing macros and all tests
 - Update doc
 
 All baselines update have been manually validated.

### CODE REVIEW

File: testing/baselines/TestInteractionAnimationCameraMovement.png
Comment (Meakk) [inline testing/baselines/TestInteractionAnimationCameraMovement.png:1]: what happened with the progress bar?

File: library/src/image.cxx [hunk 1]
```diff
@@ -295,22 +359,18 @@ bool image::compare(const image& reference, double threshold, image& diff, doubl
...
+#endif
 }
 
 //----------------------------------------------------------------------------
 bool image::operator==(const image& reference) const
 {
-  image diff;
   double error;
-  return this->compare(reference, 0, diff, error);
+  // TODO: On macOS arm vtkImageSSIM can provide non-zero error for identical images for some reason
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply library/src/image.cxx:372]: If you could try to replicate locally first thatd be great

File: testing/baselines/TestInteractionAnimationCameraMovement.png
Comment (mwestphal) [reply testing/baselines/TestInteractionAnimationCameraMovement.png:1]: The animation bar is just not at the right location in the baseline. This fixes it. TBH our interactive animation is a bit flaky, maybe we will need to reconsider these tests at some point.

File: library/src/image.cxx [hunk 2]
```diff
@@ -384,6 +446,7 @@ void image::save(const std::string& path, SaveFormat format) const
 //----------------------------------------------------------------------------
 std::vector<unsigned char> image::saveBuffer(SaveFormat format) const
 {
+  // TODO Check type is compatible
```
Comment (mwestphal) [inline library/src/image.cxx:449]: Will do in another PR

File: library/testing/CMakeLists.txt [hunk 3]
```diff
@@ -76,6 +76,13 @@ set_target_properties(libf3dSDKTests PROPERTIES
 
 target_compile_definitions(libf3dSDKTests PRIVATE F3D_MODULE_EXR=$<BOOL:${F3D_MODULE_EXR}>)
 
+# Remove this once VTK 9.3 support is removed
+if(VTK_VERSION VERSION_GREATER_EQUAL 9.3.20240729)
+  target_compile_definitions(libf3dSDKTests PRIVATE F3D_SSIM_COMPARE=1)
+else()
+  target_compile_definitions(libf3dSDKTests PRIVATE F3D_SSIM_COMPARE=0)
+endif()
```
Comment (Meakk) [inline library/testing/CMakeLists.txt:84]: Simplify with the code below and use `#ifdef F3D_SSIM_COMPARE`?
```suggestion
if(VTK_VERSION VERSION_GREATER_EQUAL 9.3.20240729)
  target_compile_definitions(libf3dSDKTests PRIVATE F3D_SSIM_COMPARE)
endif()
```

**Review by Meakk — COMMENTED** [review]
Other than the small style comment, lgtm

File: library/src/image.cxx (hunk 2 above)
Comment (mwestphal) [reply library/src/image.cxx:450]: https://github.com/f3d-app/f3d/pull/1565

**Review by Meakk — APPROVED** [review]
